### DISCUSSION TITLE
Error in agent statistics when filtering by department

### DISCUSSION THREAD

**Discussion by alonsoac** [LiveHelperChat/livehelperchat#2095] [General]
Hello. I had to add this code to avoid a SQL error when filtering by department:

lhc_web/lib/core/lhchat/lhchatstatistic.php

    public static function numberOfChatsDialogsByUserParticipant($days = 30, $filter = array(), $groupField = 'lh_chat_participant.user_id')
    {
        if ($groupField == 'transfer_uid') {
            if (isset($filter['filterin']['user_id'])) {
                $filter['filterin']['transfer_uid'] = $filter['filterin']['user_id'];
                unset($filter['filterin']['user_id']);
            }
            if (isset($filter['filterin']['lh_chat.user_id'])) {
                $filter['filterin']['lh_chat.transfer_uid'] = $filter['filterin']['lh_chat.user_id'];
                unset($filter['filterin']['lh_chat.user_id']);
            }
            if (isset($filter['filtergt']['user_id'])) {
                $filter['filtergt']['transfer_uid'] = $filter['filtergt']['user_id'];
            }
        }
        //*ADDED:*
          if (isset($filter['filterin']['dep_id'])){
                $filter['filterin']['`lh_chat_participant`.`dep_id`'] = $filter['filterin']['dep_id'];
                unset($filter['filterin']['dep_id']);
            }

**Comment by remdex** [comment]
can i get a query which get's this error? no need domain, just end part

**Reply by remdex** [reply]
And here is a proper fix... https://github.com/LiveHelperChat/livehelperchat/commit/93b27dd8430b45c3724b7876c976fac10cccc8e4
